### ISSUE TITLE
MSBuild 15.4 randomly hangs on Ubuntu 16.04

### ISSUE THREAD

**Issue by dmitrykorolevlighting** [microsoft/msbuild#2685]
I try to build the solution consisting of 20 projects on Windows and Linux in Debug and Release configuration using MSBuild command line. On Windows everything always works as expected. The same is for Linux for Debug configuration. However, Release configuration on Linux sometimes hangs indefinitely without producing any errors. In 90+% of cases it is the same project where building is stopped. However, sometimes it can be other projects. Probably, it is somehow timing related. There is nothing special about the projects on which building hangs, i.e. they are configured in the same way as other projects in the solution. I tried diffirent MSBuild parameters (e.g.  /p:UseSharedCompilation=false, /target:Rebuild, /target:Clean,Build, /maxcpucount:1, /nodeReuse:false, etc.) as well as their combination, but nothing resolved the issue. I can reproduce the issue only for Release configuration on Linux. If I retry building the solution at the same machine, it may succeed. Approximately 1 out 10 building attempts hangs.

**Comment by dmitrykorolevlighting** [comment]
UPD: ok, after a very large number of attempts, I was able to reproduce the issue on Linux with Debug configuration...

**Comment by cdmihai** [comment]
Is it possible to share the projects, command line arguments, and msbuild version / dotnet CLI version you're using?

As a workaround, can you please try turning off multi proc builds by adding the `/m:1` command line argument to msbuild?

**Comment by dmitrykorolevlighting** [comment]
Msbuild version: 15.4.0.0 (Thu Oct 12 17:13:10 UTC 2017) for Mono
All the projects in the solution use .NET Framework 4.7.

The command line parameters are quite trivial:
```msbuild /p:Configuration=Release /t:Clean,Build "pathToSolution.sln"```

I tried to extend the above line with different parameters (see my initial message) including ```/m:1```. However, with no success.

Unfortunately, I cannot share the projects. I will try to create a dummy solution and reproduce the issue there. If I succeed, then I will share it.
